src/tests/unittests/configuration_helper/test_configuration_helper.py

```diff
@@ -0,0 +1,106 @@
+import unittest
+from unittest.mock import MagicMock, Mock, patch
+
+from qilib.configuration_helper import InstrumentConfiguration, InstrumentConfigurationSet
+from qilib.configuration_helper.configuration_helper import ConfigurationHelper
+from qilib.utils import PythonJsonStructure
+from qilib.utils.storage import StorageMemory
+
+
+class TestConfigurationHelper(unittest.TestCase):
+    def test_snapshot(self):
+        storage = StorageMemory('some_name')
+        mock_configuration = MagicMock()
+        configuration_helper = ConfigurationHelper(storage, mock_configuration)
+        configuration_helper.snapshot(['some', 'tag'])
+        mock_configuration.snapshot.assert_called_once_with(['some', 'tag'])
+
+    def test_retrieve_inactive_configuration_from_storage(self):
+        tag = ['some']
+        storage = Mock()
+        configuration_tags = [['some_tag']]
+        configuration = {'adapter_class_name': 'Dummy', 'address': 'fake', 'configuration': {'param': 'value'}}
+        storage.load_data.side_effect = [configuration_tags, configuration]
+        with patch('qilib.configuration_helper.instrument_configuration.InstrumentAdapterFactory'):
+            configuration_helper = ConfigurationHelper(storage)
+            configuration_helper.retrieve_inactive_configuration_from_storage(tag)
+        inactive_configuration = configuration_helper.inactive_configuration
+        self.assertListEqual(inactive_configuration.tag, tag)
+        self.assertListEqual(inactive_configuration.instruments[0].tag, ['some_tag'])
+        self.assertEqual(inactive_configuration.instruments[0].address, 'fake')
+        self.assertDictEqual(inactive_configuration.instruments[0].configuration, {'param': 'value'})
+
+    def test_configuration_helper_integration(self):
+        storage = StorageMemory('some_name')
+        configuration_1 = PythonJsonStructure(amper=0.005)
+        configuration_2 = PythonJsonStructure(frequency='2.4 GHz')
+        with patch('qilib.configuration_helper.instrument_configuration.InstrumentAdapterFactory'):
+            instrument_1 = InstrumentConfiguration('DummyClass', 'fake-address-1', storage, tag=['instrument_1'],
```

Comment:
Because tag is defined as list[str] the tag of instrument1 can also be ['this', 'is', 'the', 'tag', 'of', 'instrument1']. Is that what is meant with this design?

Reply:
Yes, the tag is a path in a tree as described in the `StorageInterface` class.